Expensify/App
# Adds KYC Wall trigger for Transfer Balance action

cc @stitesExpensify @nickmurray47 

### Details

- Adds the KYC checks to attempts to transfer a balance
- Fixes up the positioning of the "add payment method" popover

### Fixed Issues
$ https://github.com/Expensify/Expensify/issues/187802

### Tests

**Note:** In order to simulate a user instantly passing Onfido + IDOlogy we need to take a few additional steps that are explained [here](https://stackoverflow.com/c/expensify/questions/10607).

1. Start with an account that has a gold wallet + payment method added i.e. able to "Send Money"
2. Send money to an account that does not have a gold wallet
3. As the recipient, navigate to the "Settings > Payments" screen and tap "Transfer Balance"
4. Verify that a popup appears asking you to add a "Bank account" or "Debit card"
![2022-01-20_09-36-11](https://user-images.githubusercontent.com/32969087/150410044-1f680f14-c6f9-4765-b24a-71d779e2192c.png)
5. Select Bank Account + add a test bank account
6. Verify you are navigated to the "Verify identity" flow next
![2022-01-20_09-37-03](https://user-images.githubusercontent.com/32969087/150410112-26091d3c-aa86-478b-8f41-b07c6ad964ec.png)
7. Go through Onfido 
8. Enter the required information in Additional details step (follow second answer [here](https://stackoverflow.com/c/expensify/questions/10607/10608#10608))
9. Verify that you are brought to the balance transfer page
![2022-01-20_09-43-34](https://user-images.githubusercontent.com/32969087/150410922-bede148c-69db-4d5e-b6a0-78665092bb02.png)


### QA Steps

Internal QA ... maybe Production QA? Not sure if Onfido works on staging.

### Tested On


### Screenshots


#### Web


#### Mobile Web


#### Desktop


#### iOS


#### Android
